### ISSUE TITLE
Audio Issue - Rapberry PI 3 + BT

### ISSUE THREAD

**Issue by AlexPaterson77** [SilentNightx/VitaDockPlus#42]
My Psvita dock+ 2.5 installation is on a Raspberry PI 3, connected to the original docking station by SONY via USB and to my TV using an HDMI cable.

It works perfectly with video and also connects to my dualshock 3 with the ds34Vita by MERLev plugin. 
I am having issues when connecting my vita to raspberry 3 via bluetooth in order to pass audio. The stream is perfect, through hdmi, when the console is not on my dock. 
Though: whenever i plug the vita onto my dock or USB cable (i tried two different cables, one original, the other not, with and without original dock) audio stops. I tried turning up the volume but it simply doesn’t get out, neither from the tv nor from the vita.

**Comment by VicMuuh** [comment]
It's exactly the same for me. Raspberry Pi 3 Model B Rev 1.2. 

Any advice?

**Comment by SilentNightx** [comment]
Unable to replicate but I don't have the Pi 3.

**Comment by AlexPaterson77** [comment]
Just tried version 2.6 of VitaDockPlus

On version 2.5, I could pair my vita with vitadock and stream audio through HDMI if the console was not connected to the dock.

THE FOLLOWING IS WRONG.

_**On version 2.6, I can pair my vita with vitadock, but the audio streaming does not get out of HDMI, either when the console is not on the dock and when it is on the dock.**_ 

THIS IS WRONG. I MANAGED TO STREAM AUDIO FROM MY PSVITA ONLY WHEN IT IS NOT CONNECTED VIA USB.

**In short: with 2.6, the console I connected via Bluetooth to my raspberry 3 starts being completely silent after the pairing, as if the stream is routed to something else, though no sound gets out of my TV.**

**Comment by SilentNightx** [comment]
With two reports I've decided to put it down as a potential issue with the image on Pi 3. If it works for anyone or if anyone finds a solution please update this issue.

**Comment by Pandatective** [comment]
I was curious about the Vita Dock and since I have a RPI 3 around, I gave it a try. I have tested 2.5 and 2.6 and I couldn't get the BT audio working. However, I tried the "Current" 2.0 build and BT audio did work with this one, except it's cutting out from time to time.

I'm aware it has been a while since the last message in the discussion, but I was wondering if there may be other things I could check, or it would be more preferable to use a different RPI model or use AUX instead?

**Comment by AlexPaterson77** [comment]
So, this means it’s actually an issue with Vita Dick Plus and not something related to the plugin on the ps vita. Cool to know.
Will try 2.0 as soon as I’m home

**Comment by AlexPaterson77** [comment]
I confirm:

2.5 & 2.6 do not work with the internal Bluetooth receiver of a Raspberry Pi 3.

2.0 stutters a bit, though works

**Comment by Pandatective** [comment]
To be a bit more specific, it works with "Current" 2.0 build, but not with the "Experimental" one, which the latter's main difference,  from my understanding, is based on Raspberry Pi OS Bullseye instead of Buster.

From my understanding, there's a difference between both OS when it comes to configuring the audio. Of course, I'm not too familiar with that kind of stuff and perhaps I'm looking at the wrong place, but I'll try to see if I can find something.

**Comment by Pandatective** [comment]
Sorry I took so long to provide an update; Was preoccupied with other stuff.

Turns out I didn't have to look too far. Whenever I try a build based on Bullseye, I do a right-click on the volume icon in the taskbar and then notice the output is set to  "AV Jack" instead of "HDMI" by default.

The BT audio is now working after switching it on "HDMI", but I'm noticing a delay of a second for the audio. I could blame this on the RPi3, but the delay was far less noticeable in the 2.0 "current" build despite stuttering for a bit (if I remember correctly), so I'm not sure if there's adjustments that could be done.

**Comment by AlexPaterson77** [comment]
Hi, I had already tried this solution back in time: it works until you put the psvita in the dock. If you connect it to the raspberry via usb, audio stops working after some seconds.

It seems like there’s something on the raspberry that checks the Bluetooth connections and stops the audio device once the usb is found as plugged in

**Comment by Pandatective** [comment]
On this part, I do not know. 

For me, the audio works via Bluetooth before and after plugging it on the USB and it has been running for several minutes. Speaking of the delay, I left it running a bit longer and the delay is gone, as per noted in Readme.

**Comment by AlexPaterson77** [comment]
What plugin are you using to stream from the psvita to your raspberry pi 3 ?

**Comment by Pandatective** [comment]
udcd_uvc_oled_off.skprx from https://github.com/xerpi/vita-udcd-uvc/releases/tag/1.7

**Comment by AlexPaterson77** [comment]
I managed to test a bit more, with the same plugin Pandatective is using and after disabling all non essential plugins on my PsVita.

With version 2.6:

- I connected it to three different ac adapters, to be sure it was not an undervoltage issue. The one I am using right now is not showing the Low Voltage alert the first I tested with was popping out from time to time.

- I connected the audio via internal bluetooth and played with my PsVita for a while in order to understand if the audio did drop. I started a game and the music did not drop for some minutes.

- While still in the game I connected the PsVita to my docking station: Audio works correctly before the video gets streamed to HDMI and after. Though, some seconds later, audio stops. 

- I tried with another cable, same behaviour.

I will do the same tests with version 2.0 as soon as I have some time, though the first time I tested 2.0 I managed to play with a connected DualShock 3, audio and video streaming through HDMI, for many minutes.

**Comment by AlexPaterson77** [comment]
I confirm what I was saying in my previous comment. 
With version 2.0, I did the same I did with 2.6:

- I connected the audio via internal bluetooth and played with my PsVita for a while in order to understand if the audio did drop. I started a game and the music did never drop.

- While still in the game I connected the PsVita to my docking station: Audio works correctly before the video gets streamed to HDMI and after. The audio never drops.

The PsVita is exactly the same I used for the 2.6 test, same configuration and same plugins.

**Comment by TDGalea** [comment]
Unfortunately still the same here. Gonna have to find where my 3A has hidden itself and check that.

**Comment by TDGalea** [comment]
Found my 3A and can confirm this issue is *not* present there, so this really is an issue with the single model it seems.

**Comment by AlexPaterson77** [comment]
Can you give me some details ? Build you tried, settings you tweaked….

**Comment by TDGalea** [comment]
> Can you give me some details ? Build you tried, settings you tweaked….

If directed at me, it's just the latest build as-is. Literally just moved the SD card from the 3B to 3A without any changes.

**Comment by AlexPaterson77** [comment]
@TDGalea 

So you confirm it works on 3A and not in 3B, correct ?

**Comment by TDGalea** [comment]
> @TDGalea 
> 
> So you confirm it works on 3A and not in 3B, correct ?


@AlexPaterson77
Correct.

**Comment by gfAlvaro** [comment]
Hello.

I am having this issue with AUX audio as well, and even doesn't work if I connect the cable directly to my TV which has AUX audio port. I'm using Raspberry Pi 3B+ because it is the one I already own. My TV isn't a Smart TV and because of that I needed to use Raspberry with VitaDockPlus. (The video streaming is fine.)

My PS Vita works fine in "portable mode" with headphones so the AUX port isn't the problem.

**Comment by AlexPaterson77** [comment]
I need to add: I am having this issue on 

Raspberry Pi 3 Model B Plus Rev 1.3

All RP3 Model B apparently suffer from this problem.
It would be interesting to try a build based on Bullseye, with all the 2.6 script updates.

**Comment by zenaror** [comment]
Same issue on my Raspberry Pi 3 Model B rev 1.2, with the latest image

Tested with main version 2.0 works fine.besides the micro stuttering on sound sometimes
